pkg/harvester/edit/harvesterhci.io.virtualmachineimage.vue

```diff
@@ -269,6 +271,23 @@ export default {
       }
     },
 
+    async handleEditImage(buttonCb) {
+      try {
+        const data = [{
+          op: 'replace', path: '/metadata/labels', value: this.value.metadata.labels
+        }, {
+          op: 'replace', path: '/metadata/annotations', value: this.value.metadata.annotations
+        }];
+
+        await this.value.patch(data);
```

Comment:
The issue occurs because `this.save` uses a `PUT` request, and misses the `status` in request body, it also overwrites progress if the user edits the config before the upload finishes.
Using `PATCH` fixes this by only updating the necessary fields.